otterdog/operations/check_status.py

```diff
@@ -0,0 +1,126 @@
+#  *******************************************************************************
+#  Copyright (c) 2023-2025 Eclipse Foundation and others.
+#  This program and the accompanying materials are made available
+#  under the terms of the Eclipse Public License 2.0
+#  which is available at http://www.eclipse.org/legal/epl-v20.html
+#  SPDX-License-Identifier: EPL-2.0
+#  *******************************************************************************
+
+from __future__ import annotations
+
+from sys import stderr as sys_stderr
+from json import dumps as json_dumps
+from typing import TYPE_CHECKING
+
+from .diff_operation import DiffOperation, DiffStatus
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from otterdog.config import OrganizationConfig, OtterdogConfig
+    from otterdog.jsonnet import JsonnetConfig
+    from otterdog.models import LivePatch, ModelObject
+    from otterdog.models.github_organization import GitHubOrganization
+    from otterdog.utils import Change, IndentingPrinter
+
+    from .validate import ValidationStatus
+
+
+class CheckStatusOperation(DiffOperation):
+    """
+    Check the status of current configuration (validity and wether it is in sync with the GitHub live configuration)
+    """
+
+    def __init__(
+        self,
+        no_web_ui: bool,
+        repo_filter: str,
+    ):
+        super().__init__(no_web_ui, repo_filter, False, False, "*")
+        self.orgs_status: list[Any] = []
+
+    def init(self, config: OtterdogConfig, printer: IndentingPrinter) -> None:
+        super().init(config, printer)
+
+    def pre_execute(self) -> None: ...
+
+    def post_execute(self) -> None:
+        print(json_dumps(self.orgs_status, indent=2))
+
+    def resolve_secrets(self) -> bool:
+        return False
+
+    async def handle_finish(
+        self, org_id: str, diff_status: DiffStatus, validation_status: ValidationStatus, patches: list[LivePatch]
+    ) -> int:
+        is_archived = await self.gh_client.rest_api.org.is_archived(org_id)
+        org_status = {
+            "org_id": org_id,
+            "is_archived": is_archived,
+            "validation_status": {
+                "is_valid": validation_status.total_notices() == 0,
+                "infos": validation_status.infos,
+                "warnings": validation_status.warnings,
+                "errors": validation_status.errors,
+            },
+            "sync_status": {
+                "in_sync": validation_status.total_notices() == 0 and not patches,
+                "additions": diff_status.additions,
+                "changes": diff_status.differences,
+                "deletions": diff_status.deletions,
+            },
+        }
+        self.orgs_status.append(org_status)
+        print(f"archived: {is_archived}", file=sys_stderr)
```

Comment:
why do you use print to stderr and not the builtin self.printer?

Reply:
I want to separate progress information (informative messages, to stderr) from actual output (JSON content, to stdout). This way, I can redirect stdout to a file while still seeing progress updates. It's especially useful when the command is run without specifying an organization, i.e., it should execute for all organizations.

Ideally, the JSON output should be optional and written to a file provided as a CLI argument. I will give a try if you feel that it would be cleaner.

Reply:
I came up with other operations that do something similar.

Will review properly, the problem with print statements is that the linter will complain.

Reply:
I've changed the logic a little bit. There is now a `--json` argument to the `check-status` command for the json output. With that, I can now use the regular `self.printer` for progress/info logging.